File: frontend/de/unika/ipd/grgen/ast/type/basic/UntypedExecVarTypeNode.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 6.0
 * Copyright (C) 2003-2021 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

/**
 * @author Edgar Jakumeit
 */

package de.unika.ipd.grgen.ast.type.basic;

import de.unika.ipd.grgen.ast.BaseNode;
import de.unika.ipd.grgen.ast.type.TypeNode;
import de.unika.ipd.grgen.ir.IR;
import de.unika.ipd.grgen.ir.type.basic.UntypedExecVarType;

import java.util.Collection;
import java.util.Vector;

public class UntypedExecVarTypeNode extends BasicTypeNode
{
	static {
		setName(UntypedExecVarTypeNode.class, "untyped exec variable type");
	}

	// TODO: No instance is ever used! Probably useless...
	public static class Value
	{
		public static Value NULL = new Value() {
			@Override
			public String toString()
			{
				return "Untyped null";
			}
		};

		private Value()
		{
		}
	}

	public UntypedExecVarTypeNode()
	{
	}

	/** returns children of this node */
	@Override
	public Collection<BaseNode> getChildren()
	{
		Vector<BaseNode> children = new Vector<BaseNode>();
		// no children
		return children;
	}

	/** returns names of the children, same order as in getChildren */
	@Override
	public Collection<String> getChildrenNames()
	{
		Vector<String> childrenNames = new Vector<String>();
		// no children
		return childrenNames;
	}

	@Override
	public boolean isCompatibleTo(TypeNode t)
	{
		// compatible to everything
		return true;
	}

	@Override
	public boolean isCastableTo(TypeNode t)
	{
		return isCompatibleTo(t);
	}

	@Override
	protected IR constructIR()
	{
		return new UntypedExecVarType(getIdentNode().getIdent());
	}

	@Override
	public String toString()
	{
		return "untyped";
	}
}

File: frontend/de/unika/ipd/grgen/ast/type/basic/UntypedExecVarTypeNode.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

/// <summary>
/// @author Edgar Jakumeit
/// </summary>

namespace de.unika.ipd.grgen.ast.type.basic
{
	using System.Collections.Generic;

	using BaseNode = de.unika.ipd.grgen.ast.BaseNode;
	using TypeNode = de.unika.ipd.grgen.ast.type.TypeNode;
	using IR = de.unika.ipd.grgen.ir.IR;
	using UntypedExecVarType = de.unika.ipd.grgen.ir.type.basic.UntypedExecVarType;


	public class UntypedExecVarTypeNode : BasicTypeNode
	{
		static UntypedExecVarTypeNode()
		{
			SetClassName(typeof(UntypedExecVarTypeNode), "untyped exec variable type");
		}

		// TODO: No instance is ever used! Probably useless...
		public class Value
		{
			public static Value NULL = new ValueAnonymousInnerClass();

			private class ValueAnonymousInnerClass : Value
			{
				public override string ToString()
				{
					return "Untyped null";
				}
			}

			internal Value()
			{
			}
		}

		public UntypedExecVarTypeNode()
		{
		}

		/// <summary>
		/// returns children of this node </summary>
		public override ICollection<BaseNode> Children
		{
			get
			{
				IList<BaseNode> children = new List<BaseNode>();
				// no children
				return children;
			}
		}

		/// <summary>
		/// returns names of the children, same order as in getChildren </summary>
		public override ICollection<string> ChildrenNames
		{
			get
			{
				IList<string> childrenNames = new List<string>();
				// no children
				return childrenNames;
			}
		}

		public override bool IsCompatibleTo(TypeNode t)
		{
			// compatible to everything
			return true;
		}

		public override bool IsCastableTo(TypeNode t)
		{
			return IsCompatibleTo(t);
		}

		protected internal override IR ConstructIR()
		{
			return new UntypedExecVarType(Ident.IRIdent);
		}

		public override string ToString()
		{
			return "untyped";
		}
	}

}
